controllers/factory/alertmanager.go

```diff
@@ -415,27 +449,25 @@ func makeStatefulSetSpec(cr *victoriametricsv1beta1.VMAlertmanager, c *config.Ba
 		TerminationMessagePolicy: v1.TerminationMessageFallbackToLogsOnError,
 	}
 	vmaContainer = buildProbe(vmaContainer, cr) // cr.Spec.EmbeddedProbes, healthPath, cr.Spec.PortName, true)
-	defaultContainers := []v1.Container{
-		vmaContainer,
-		{
-			Name:  "config-reloader",
-			Image: fmt.Sprintf("%s", formatContainerImage(c.ContainerRegistry, c.VMAlertManager.ConfigReloaderImage)),
-			Args: []string{
-				fmt.Sprintf("-webhook-url=%s", localReloadURL),
-				fmt.Sprintf("-volume-dir=%s", alertmanagerConfDir),
-			},
-			VolumeMounts: []v1.VolumeMount{
-				{
-					Name:      "config-volume",
-					ReadOnly:  true,
-					MountPath: alertmanagerConfDir,
-				},
-			},
-			Resources:                resources,
-			TerminationMessagePolicy: v1.TerminationMessageFallbackToLogsOnError,
+	configReloaderContainer := v1.Container{
+		Name:  "config-reloader",
+		Image: fmt.Sprintf("%s", formatContainerImage(c.ContainerRegistry, c.VMAlertManager.ConfigReloaderImage)),
+		Args: []string{
+			fmt.Sprintf("-webhook-url=%s", localReloadURL),
+			fmt.Sprintf("-volume-dir=%s", alertmanagerConfDir),
 		},
+		VolumeMounts:             crVolumeMounts,
+		Resources:                resources,
+		TerminationMessagePolicy: v1.TerminationMessageFallbackToLogsOnError,
+	}
+	if len(cr.Spec.ConfigMaps) > 0 {
```

Comment:
does config-reloader recursively watches dir changes? probably we have to add TemplatesDir with cm name?
 
 As far as I know it uses fs.notify

```Are subdirectories watched too?
No, you must add watches for any directory you want to watch (a recursive watcher is on the roadmap: https://github.com/fsnotify/fsnotify/issues/18).
```

Reply:
It worked fine during testing, but it may have happened by accident, so I'll check carefully.

Reply:
For some reason everything works for me, but you're right, it shouldn't work (it can depend on the environment), so i fixed